Allow the same behavior with promise as without

nicolasayotte opened the pull request:
The missing response information in the promise version breaks the symmetry of the code between the promise and callback versions. For the promise, I recommend returning the body and response as an object so that this is rectified.

This is a **breaking** change for people already using the promise version of the code.

Commit:
Allow the same behavior with promise as without

The lack of the response information in the promise version breaks the symmetry of the code between the promise and callback versions. For the promise, I suggest returning the body and response as an object so that this is rectified.

brycebaril commented:
Hi @nicolasayotte! Thanks for another submission! This one has some tests that need to be fixed, when I run the tests with this PR I get three failures; patch those up and I'll take another look 👍

brycebaril commented:
Oh, also the README will need to be updated

Commit:
Updated the readme and the tests for the new promise result object

Fixed and added some tests to ensure the same behaviour between the callback and promise version of the module.
Updated the readme.md file to reflect this.

nicolasayotte commented:
Sorry, you're absolutely right! I have updated the tests and verified that they all pass.
I have also updated the README.md file to show how to use the result object and a quick ES2016 use of the promise in the async/await pattern (also valid in Typescript).

brycebaril commented on README.md:
```diff
@@ -47,9 +47,29 @@ var req = request(options, function callback(err, response, body) {
+  if (result.response.statusCode === 201) {
+    console.log(result.response.headers.location);
+  } else {
+    console.log(result.body);
```
Some semicolons got into the README examples, which mostly only matters for consistency; probably best to have consistent style throughout.

brycebaril commented on promise.js:
```diff
@@ -13,7 +13,11 @@ function requestPromise(opts) {
+          body: body,
+          response: response
+        };
+        resolve(result);
```
remove semicolons :)

nicolasayotte replied:
Ha ha, alright :) One second

brycebaril reviewed (changes requested):
Couple minor nits about semicolons and it should be good to go!

Commit:
No Semicolons for consistent style

nicolasayotte commented:
I delete those semi-colons. I code in Typescript mostly so I add them out of habit. :)

brycebaril commented:
Awesome, thanks! Published as v2.0.0

nicolasayotte commented:
Thank you. 👍
